dask_cuda/proxify_device_objects.py

```diff
@@ -12,6 +12,8 @@
 dispatch = Dispatch(name="proxify_device_objects")
 ignore_types = None
 
+T = TypeVar("T")
```

Comment:
Sorry for the naive question, but what does `TypeVar("T")` mean? It looks like a `template<class T>` in C++, but I'm not sure I understand why that makes sense here and why it's better than `Any`.

Reply:
It is basically to say "the input type equals to output type" in functions like:
```python
def unproxify_device_objects(obj: T, skip_explicit_proxies: bool = False) -> T:
```
This way, type checkers like `mypy` can infer the output type of `unproxify_device_objects()`

Reply:
Ah sure, that makes sense. Thanks for the clarification!